ci_cd/tasks/update_deps.py

```diff
@@ -365,7 +377,10 @@ def update_deps(  # pylint: disable=too-many-branches,too-many-locals,too-many-s
                     current_version = specifier.version.split(".")
                     break
             else:
```

Comment:
The handling of the epoch is a bit confusing. The else clause seems inverted; if there is no epoch (epoch == 0), why prepend it to 0.0.0? This seems like a potential error or at least a place where the intent isn't clear. Clearer comments or a refactor for readability might be needed.